Common/Input/SDL/SDL_InputManager.cpp

```diff
@@ -0,0 +1,207 @@
+		mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(SDLK_x, MyGUI::KeyCode::X));
+		mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(SDLK_c, MyGUI::KeyCode::C));
+		mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(SDLK_v, MyGUI::KeyCode::V));
+		mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(SDLK_b, MyGUI::KeyCode::B));
+		mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(SDLK_n, MyGUI::KeyCode::N));
+		mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(SDLK_m, MyGUI::KeyCode::M));
+		mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(SDLK_COMMA, MyGUI::KeyCode::Comma));
+		mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(SDLK_PERIOD, MyGUI::KeyCode::Period));
+		mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(SDLK_SLASH, MyGUI::KeyCode::Slash));
+		mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(SDLK_RSHIFT, MyGUI::KeyCode::RightShift));
+		//mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(SDLK_, MyGUI::KeyCode::Multiply));
+		mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(SDLK_LALT, MyGUI::KeyCode::LeftAlt));
+		mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(SDLK_SPACE, MyGUI::KeyCode::Space));
+		mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(SDLK_CAPSLOCK, MyGUI::KeyCode::Capital));
+		//mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(, SC_F));
+		//mSDLVKeyMap.insert(std::pair<int, MyGUI::KeyCode>(, SC_F));
+	}
+
+	void InputManager::buildMouseButtonMap()
+	{
+		mSDLMouseMap.clear();
+		mSDLMouseMap.insert(std::pair<int, MyGUI::MouseButton>(SDL_BUTTON_LEFT, MyGUI::MouseButton::Left));
+		mSDLMouseMap.insert(std::pair<int, MyGUI::MouseButton>(SDL_BUTTON_RIGHT, MyGUI::MouseButton::Right));
+		mSDLMouseMap.insert(std::pair<int, MyGUI::MouseButton>(SDL_BUTTON_MIDDLE, MyGUI::MouseButton::Middle));
+	}
+
+	void InputManager::createInput()
+	{
+		MyGUI::Gui::getInstance().eventFrameStart += MyGUI::newDelegate(this, &InputManager::frameEvent);
+	}
+
+	void InputManager::destroyInput()
+	{
+
+	}
+
+	void InputManager::updateCursorPosition()
+	{
+	}
+
+	void InputManager::frameEvent(float _time)
+	{
+		computeMouseMove();
+	}
+
+	void InputManager::computeMouseMove()
+	{
+		if (mMouseMove)
+		{
+			injectMouseMove(mMouseX, mMouseY, mMouseZ);
+			mMouseMove = false;
+		}
+	}
+
+	bool InputManager::mouseMoved(SDL_MouseMotionEvent &evt)
+	{
+		mMouseX = evt.x;
+		mMouseY = evt.y;
+		mMouseMove = true;
+		return true;
+	}
+
+	bool InputManager::mousePressed(SDL_MouseButtonEvent &evt)
+	{
+		computeMouseMove();
+		injectMousePress(mMouseX, mMouseY, mSDLMouseMap[evt.button]);
+		return true;
+	}
+
+	bool InputManager::mouseReleased(SDL_MouseButtonEvent &evt )
+	{
+		computeMouseMove();
+		injectMouseRelease(mMouseX, mMouseY, mSDLMouseMap[evt.button]);
+		return true;
+	}
+
+	bool InputManager::keyPressed(SDL_Keycode &key, const char* text)
+	{
+		if (mSDLVKeyMap.count(key) == 0) {
+			return false;
+		}
+		MyGUI::KeyCode myGuiKeyCode = mSDLVKeyMap[key];
+		//const char* ch = SDL_GetKeyName(key.keysym.sym);
+		if (text == NULL) {
```

Comment:
This is incorrect. When ```text``` is NULL, we will already have injected the text via the SDL_TEXTINPUT event. Or should, anyway. Is SDL_TEXTINPUT working at all? I don't see a SDL_StartTextInput anywhere in the code, which is needed to start receiving textinput events.